src/app/3d/qgs3dmapcanvasdockwidget.cpp

```diff
@@ -291,6 +297,34 @@ void Qgs3DMapCanvasDockWidget::configure()
   mBtnMapThemes->setDisabled( map->terrainGenerator()->type() == QgsTerrainGenerator::Mesh );
 }
 
+void Qgs3DMapCanvasDockWidget::exportScene()
+{
+
+  QDialog dlg;
+  dlg.setWindowTitle( tr( "Export 3D Scene" ) );
+  dlg.setObjectName( QStringLiteral( "3DSceneExportDialog" ) );
+  dlg.setMinimumSize( 380, 460 );
+//  QgsGui::instance()->enableAutoGeometryRestore( &dlg );
+
+  Qgs3DMapExportSettings *exportSettings = new Qgs3DMapExportSettings( &dlg );
+  QgsMap3DExportWidget *w = new QgsMap3DExportWidget( mCanvas->scene(), exportSettings, &dlg );
+
+  QDialogButtonBox *buttons = new QDialogButtonBox( QDialogButtonBox::Ok | QDialogButtonBox::Cancel, &dlg );
+
+  connect( buttons, &QDialogButtonBox::accepted, &dlg, &QDialog::accept );
+  connect( buttons, &QDialogButtonBox::rejected, &dlg, &QDialog::reject );
+  connect( &dlg, &QDialog::accepted, [ = ]()
+  {
+    w->exportScene();
+  } );
```

Comment:
It would be easier to read if instead of having this slot with a lambda function the `w->exportScene()` call would be on the last line of this function... because if dlg.exec() returns true, that means the dialog has been accepted.

Reply:
For some reason I thought that we will want the dialog to not block the whole app in the future so I made it like that